services/streammanager/googlesheets/googlesheetsmanager.go

```diff
@@ -161,46 +159,93 @@ func insertDataToSheet(sheetsClient *sheets.Service, spreadSheetId string, sprea
 		return fmt.Errorf("[GoogleSheets] error  :: Failed to initialize google-sheets client")
 	}
 
-	if isHeader {
-		_, err = sheetsClient.Spreadsheets.Values.Update(spreadSheetId, spreadSheetTab+"!A1", &vr).ValueInputOption("RAW").Do()
+	_, err = sheetsClient.Spreadsheets.Values.Update(spreadSheetId, spreadSheetTab+"!A1", &vr).ValueInputOption("RAW").Do()
+
+	return err
+}
+
+// Wrapper func to append row data list,
+// Returns error for failure cases of API calls otherwise returns nil
+func insertRowDataToSheet(sheetsClient *sheets.Service, spreadSheetId string, spreadSheetTab string, dataList [][]interface{}) error {
+	// Creating value range for inserting row into sheet
+	var vr sheets.ValueRange
+	vr.MajorDimension = "ROWS"
+	vr.Range = spreadSheetTab + "!A1"
+	vr.Values = dataList
+	var err error
 
-	} else {
-		_, err = sheetsClient.Spreadsheets.Values.Append(spreadSheetId, spreadSheetTab+"!A1", &vr).ValueInputOption("RAW").Do()
+	if sheetsClient == nil {
+		return fmt.Errorf("[GoogleSheets] error  :: Failed to initialize google-sheets client")
 	}
+
+	_, err = sheetsClient.Spreadsheets.Values.Append(spreadSheetId, spreadSheetTab+"!A1", &vr).ValueInputOption("RAW").Do()
+
 	return err
 }
 
 // Method to return array of values from a json.
 // source is the json object from transformer and we are iterating the json as a map
 // and we are storing the data into designated position in array based on transformer
 // mappings.
-// Example payload we have from transformer:
+// Example payload we have from transformer without batching:
 // {
 //		message:{
 //			1: { attributeKey: "Product Purchased", attributeValue: "Realme C3" }
 //			2: { attributeKey: "Product Value, attributeValue: "5900"}
 //			..
 // 		}
 // }
-func parseTransformedData(source gjson.Result) ([]string, error) {
-	messagefields := source.Get("message")
-	values := make([]string, len(messagefields.Map()))
-	var pos int
-	var err error
-	if messagefields.IsObject() {
-		for k, v := range messagefields.Map() {
-			pos, err = strconv.Atoi(k)
-			if err != nil {
-				return values, err
+// Example Payload we have from transformer with batching:
+// {
+// 		batch:[
+//			{
+//				message: {
+//					1: { attributeKey: "Product Purchased", attributeValue: "Realme C3" }
+//					2: { attributeKey: "Product Value, attributeValue: "5900"}
+//					..
+// 				}
+//			},
+//			{
+//				message: {
+//					1: { attributeKey: "Product Purchased", attributeValue: "Realme C3" }
+//					2: { attributeKey: "Product Value, attributeValue: "5900"}
+//					..
+// 				}
+//			}
+// 		]
+// }
+func parseTransformedData(source gjson.Result) ([][]interface{}, error) {
+	batch := source.Get("batch")
+	messages := batch.Array()
+	if messages != nil && len(messages) == 0 {
```

Comment:
According to the doc:
> Array returns back an array of values. If the result represents a null value or is non-existent, then an empty array will be returned. If the result is not a JSON array, the return value will be an array containing one result.

Based on the above the `nil` check appears to be redundant

Reply:
This is adressed